web/META-INF/resources/js/controllers.js

```diff
@@ -0,0 +1,261 @@
+            }
+
+            word = searchKeyWords;
+
+            if(!$rootScope.searchOptions) {
+                $rootScope.searchOptions = {};
+                opt = 'Or';
+            }
+            else
+                opt = $rootScope.searchOptions.opt;
+
+            SearchOptions.setSearchOptions({'query':word, 'opt':opt});
+            $rootScope.searchOptions.query = searchKeyWords;
+            $rootScope.searchOptions.opt = opt;
+        }
+
+        function initController() {
+            vm.setPage(1);
+        }
+
+        function setPage(page) {
+            if (page < 1 || page > vm.pager.totalPages) {
+                return;
+            }
+
+            // get pager object from service
+            vm.pager = PagerService.GetPager(vm.PDItems.length, page);
+
+            // get current page of items
+            vm.items = vm.PDItems.slice(vm.pager.startIndex, vm.pager.endIndex + 1);
+        }
+
+        this.searchTopic = function (topic) {
+            var searchKeyWords = String();
+
+            if (topic.search(',') !== -1) {
+                var topics = topic.split(',');
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            } else {
+                searchKeyWords = topic;
+            }
+
+            if (searchKeyWords.search('/') !== -1) {
+                var topics = searchKeyWords.split('/');
+                searchKeyWords = "";
+                for (var i = 0; i < topics.length; i++) {
+                    searchKeyWords += topics[i];
+                }
+            }
+
+            $rootScope.searchOptions.query = searchKeyWords;
+            MetaData.get({query: searchKeyWords, operator: opt},
+                function success(response) {
+                    vm.PDItems = response.PDResults;
+                    vm.totalMatches = vm.PDItems.length;
+                    vm.query = word;
+                    vm.opt = opt;
+                    initController();
+
+                },
+                function error(errorResponse) {
+                    console.log("Error:" + JSON.stringify(errorResponse));
+                }
+            );
+        }
+
+        MetaData.get({query: word, operator: opt},
+            function success(response) {
+                vm.PDItems = response.PDResults;
+                vm.totalMatches = vm.PDItems.length;
+                vm.query = word;
+                vm.opt = opt;
+                initController();
+
+            },
+            function error(errorResponse) {
+                console.log("Error:" + JSON.stringify(errorResponse));
+            }
+        );
+    }]);
+
+mudrodControllers.controller('datasetViewCtrl', ['$rootScope', '$scope', '$routeParams', 'DatasetDetail', 'SearchOptions',
+    function datasetViewCtrl($rootScope, $scope, $routeParams, DatasetDetail, SearchOptions) {
+        var shortname = $routeParams.shortname;
+
+        var query = String();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "query" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1854)